diff --git a/CMakeLists.txt b/CMakeLists.txt
index a76718f..ebf5ddf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,12 @@ find_package(Eigen3 REQUIRED CONFIG)
 find_package(fmt REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(tcb_span REQUIRED)
-find_package(tl-expected REQUIRED)
+find_package(tl_expected REQUIRED)
+# Alias tl_expected::tl_expected to tl::expected if needed
+if(TARGET tl_expected::tl_expected AND NOT TARGET tl::expected)
+  add_library(tl::expected INTERFACE IMPORTED)
+  set_target_properties(tl::expected PROPERTIES INTERFACE_LINK_LIBRARIES tl_expected::tl_expected)
+endif()
 
 option(RSL_ENABLE_WARNINGS "Enable compiler warnings" OFF)
 if(RSL_ENABLE_WARNINGS AND CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
diff --git a/cmake/rsl-config.cmake b/cmake/rsl-config.cmake
index b8a94cf..6a08dbd 100644
--- a/cmake/rsl-config.cmake
+++ b/cmake/rsl-config.cmake
@@ -4,7 +4,12 @@ find_dependency(Eigen3 CONFIG)
 find_dependency(fmt)
 find_dependency(rclcpp)
 find_dependency(tcb_span)
-find_dependency(tl-expected)
+find_dependency(tl_expected)
+# Alias tl_expected::tl_expected to tl::expected if needed
+if(TARGET tl_expected::tl_expected AND NOT TARGET tl::expected)
+  add_library(tl::expected INTERFACE IMPORTED)
+  set_target_properties(tl::expected PROPERTIES INTERFACE_LINK_LIBRARIES tl_expected::tl_expected)
+endif()
 
 include(${CMAKE_CURRENT_LIST_DIR}/rsl-targets.cmake)
 
